pennylane/ops/op_math/pow.py

```diff
@@ -341,8 +342,17 @@ def generator(self):
     def pow(self, z):
         return [Pow(base=self.base, z=self.z * z)]
 
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_adjoint(self):
+        return isinstance(self.z, int)
```

Comment:
Unless this has unforseen consequences, I second this comment.

Reply:
@EmilianoG-byte @albi3ro 
I think this is a more general conceptual question. Other functions of `Pow` use `isinstance(z, int)` and not `float(z).is_integer()`, so I stuck to that. If we want to change this, we should do it conciously and consistently across the code base.

Personally, I find it more appealing to have a function that behaves dependent on data types rather than dependent on the values of one and the same type. As an example: A user could use a trainable power, in principle, initialize it to `1.0` and get an error computing the _second_ function evaluation after updating it. That might be a confusing experience. In the end, I'd also be fine to switch, but this might be a bigger topic?